esmvaltool/preprocessor/_regrid.py

```diff
@@ -194,9 +195,14 @@ def regrid(src_cube, target_grid, scheme):
                 src_cube.remove_coord(coord)
 
     # Perform the horizontal regridding.
-    result = src_cube.regrid(target_grid, horizontal_schemes[scheme])
-
-    return result
+    results = iris.cube.CubeList()
+    regridder = None
+    for map_slice in src_cube.slices(['latitude', 'longitude']):
+        if not regridder:
+            regridder = horizontal_schemes[scheme].regridder(map_slice,
+                                                             target_grid)
+        results.append(regridder(map_slice))
```

Comment:
@jvegasbsc You could slightly tweak this as follows...
```python
regridder = horizontal_scheme[scheme].regridder(src_cube, target_grid)
for map_slice in src_cube.slices(['latitude', 'longitude']):
    results.append(regridder(map_slice))
```
or
```python
results = [regridder(map_slice) for map_slice in src_cube.slices(['latitude', 'longitude'])]
```

Reply:
Thanks Bill !